### DISCUSSION TITLE
Direct matrix inversion VS Featherstone's Articulated-Body Algorithm

### DISCUSSION THREAD

**Discussion by yzd0014** [google-deepmind/mujoco#1460] [General]
[fastSPD.pdf](https://github.com/google-deepmind/mujoco/files/14441082/fastSPD.pdf)
According to this paper, there are two ways to formulate equations of motions for a kinematic tree, direct matrix inversion and ABA. I wonder if mojoco is using one of those two?
It seems like mujoco is using a method that is closer to direct matrix inversion since qM is nMx1. For ABA, I think it will end up with multiple small matrices instead of a single sparse matrix. Any more information would be really appreciated!

**Comment by yuvaltassa** [comment]
Yes, we use the former.

**Reply by yzd0014** [reply]
Thank you for confirming it!
